doc/examples/type_hints.rst

```diff
@@ -0,0 +1,149 @@
+
+.. _type_hints-example:
+
+Type Hints
```

Comment:
Are there any common mypy/typing errors that users will likely run into? I suggest we add a section that shows examples of each error and how to resolve it. For example "a type for :class:`~pymongo.mongo_client.MongoClient` must be specified" error would be a good one to start with.

Reply:
Done